Mage.Sets/src/mage/cards/t/ThePrideOfHullClade.java

```diff
@@ -0,0 +1,151 @@
+
+        // {2}{U}{U}: Until end of turn, target creature you control gets +1/+0, gains "Whenever this creature deals combat damage to a player, draw cards equal to its toughness," and can attack as though it didn't have defender.
+        Ability ability = new SimpleActivatedAbility(
+                new BoostTargetEffect(1, 0)
+                        .setText("Until end of turn, target creature you control gets +1/0"),
+                new ManaCostsImpl<>("{2}{U}{U}"))
+                ;
+        ability.addEffect(
+                new GainAbilityTargetEffect(
+                        new DealsCombatDamageToAPlayerTriggeredAbility(
+                                new ThePrideOfHullCladeEffect(), false))
+                        .setText(", gains \"Whenever this creature deals combat damage to a player, draw cards equal to its toughness,\""));
+        ability.addEffect(
+                new CanAttackAsThoughItDidntHaveDefenderTargetEffect(Duration.EndOfTurn)
+                        .setText("and can attack as though it didn't have defender."));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private ThePrideOfHullClade(final ThePrideOfHullClade card) {
+        super(card);
+    }
+
+    @Override
+    public ThePrideOfHullClade copy() {
+        return new ThePrideOfHullClade(this);
+    }
+}
+
+class ThePrideOfHullCladeCostReductionEffect extends CostModificationEffectImpl {
+
+    ThePrideOfHullCladeCostReductionEffect() {
+        super(Duration.WhileOnStack, Outcome.Benefit, CostModificationType.REDUCE_COST);
+        staticText = "This spell costs {X} less to cast, where X is the total toughness of creatures you control";
+    }
+
+    private ThePrideOfHullCladeCostReductionEffect(final ThePrideOfHullCladeCostReductionEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ThePrideOfHullCladeCostReductionEffect copy() {
+        return new ThePrideOfHullCladeCostReductionEffect(this);
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && abilityToModify.getSourceId().equals(source.getSourceId())
+                && game.getCard(abilityToModify.getSourceId()) != null;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        int reductionAmount = game.getBattlefield()
+                .getAllActivePermanents(
+                        StaticFilters.FILTER_PERMANENT_CREATURE, abilityToModify.getControllerId(), game)
+                .stream()
+                .map(Permanent::getToughness)
+                .mapToInt(MageInt::getValue)
+                .sum();
+        CardUtil.reduceCost(abilityToModify, Math.max(0, reductionAmount));
+        return true;
+    }
+}
+
+class ThePrideOfHullCladeEffect extends OneShotEffect {
```

Comment:
can this be `DrawCardSourceControllerEffect` with `SourcePermanentToughnessValue` ?